nifi-registry-core/nifi-registry-data-model/src/main/java/org/apache/nifi/registry/flow/VersionedParameterContext.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.registry.flow;
+
+import io.swagger.annotations.ApiModelProperty;
+
+import java.util.Set;
+
+public class VersionedParameterContext {
+
```

Comment:
I'm ok with this data model, but @mcgilman pointed out to me today that in NiFi Parameter Contexts will probably also have an `id` field in addition to `name` to use internally. If so, it might be worth including that in the Registry model as well.

Reply:
I think since the context on the registry side is just template of the parameters (i.e. not a version controlled thing on its own), that we are ok with just a name. The NiFi side will definitely have an id so that contexts can be renamed, but that id is only relevant to the local NiFi. When you save a flow with a param context, and import to the next environment for the first time, it would create a new context with a new local id and the same name from the flow snapshot.